mass_mailing_partner/hooks.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# © 2015 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# © 2015 Antonio Espinosa <antonioea@antiun.com>
+# © 2015 Javier Iniesta <javieria@antiun.com>
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from openerp import api, SUPERUSER_ID
+
+_logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
+    with api.Environment.manage():
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        # ACTION 1: Match existing contacts
+        contact_model = env['mail.mass_mailing.contact']
+        partner_model = env['res.partner']
+        contacts = contact_model.search([('email', '!=', False)])
```

Comment:
Include `with_context(active_test=False)` for including deactivated partners (if they have e-mail)